samples/src/main/synapse/pre-process-json-2-ADX.ipynb

```diff
@@ -0,0 +1,454 @@
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Data pre-processing for Azure Data Explorer\r\n",
+        "\r\n",
+        "<img src=\"https://github.com/Azure/azure-kusto-spark/raw/master/kusto_spark.png\" style=\"border: 1px solid #aaa; border-radius: 10px 10px 10px 10px; box-shadow: 5px 5px 5px #aaa\"/>\r\n",
+        "\r\n",
+        "We often see customer scenarios where historical data has to be migrated to Azure Data Explorer (ADX). Although ADX has very powerful data-transformation capabilities via [update policies](https://docs.microsoft.com/azure/data-explorer/kusto/management/updatepolicy), sometimes more or less complex data engineering tasks must be done upfront. This happens if the original data structure is too complex or just single data elements being too big, hitting data explorer limits of dynamic columns of 1 MB or maximum ingest file-size of 1 GB for uncompressed data (see also [Comparing ingestion methods and tools](https://docs.microsoft.com/azure/data-explorer/ingest-data-overview#comparing-ingestion-methods-and-tools)) .\r\n",
```

Comment:
I don't see how this is related to historical data 
And if so lets provide an example of using creationTime - which alters the extent creation time so that the date used for retention policy changes

Reply:
whats the issue

Reply:
temp table created has a softdelete period of 7d :
Creation time is set to '8/6/2021 12:00:00 AM', which is older than the soft delete period for table 'sparkTempTable_Databricks_Shell_test2_bd28c7dd_bff4_43e6_aac8_e07b19b0c1da' ('7.00:00:00')
Is there a way to change this?

Reply:
... so not creating the retention policy with the default DefaultCleaningIntervall:

engineClient.execute(database, generateTableAlterRetentionPolicy(tmpTableName,
      DurationFormatUtils.formatDuration(writeOptions.autoCleanupTime.toMillis, durationFormat, true),
      recoverable = false), crp)

think this is causing the issue.